### PR TITLE
Ayr 486/ayr login page

### PR DESCRIPTION
**Pull request by colinbowen**

### CODE REVIEW

**Review by TomJKing — COMMENTED** [review]
Looks good. Thanks. The logic of the overriding needs to be inverted. The TDR specific styles / classes are now the exception with the new service neutral designs, and will ultimately need to be deleted. What's identified as AYR is in fact the new service neutral design and will be what is used going forward so should not split out / identified as product specific.

This does mean more code changes now, but will get the code into a state that once these designs are fully released for the product cleaning up the old TDR specific code will be much more straight forward.

File: themes/tdr/login/login-reset-password.ftl [hunk 1]
```diff
@@ -1,9 +1,14 @@
 <#import "template.ftl" as layout>
 <#assign passwordResetPageTitle = msg("passwordReset")>
```
[the remark this answers is not in the document]
Comment (colinbowen) [reply themes/tdr/login/login-reset-password.ftl:3]: Renamed to tdr specific variable

File: themes/tdr/css-src/sass/_header.scss [hunk 2]
```diff
@@ -19,7 +19,84 @@ $govuk-header-link: govuk-colour('white');
...
+  }
+
+  @include on-mobile {
+    margin-right: 0;
+    font-size: 0.8rem;
+    display: inline-block;
+  }
+}
+
+.tna-header {
```
Comment (colinbowen) [inline themes/tdr/css-src/sass/_header.scss:48]: @TomJKing Is .tna ok as a prefix for these classes. If anything is preferred we can use that instead

File: themes/tdr/login/messages/messages_en.properties [hunk 3]
```diff
@@ -43,3 +43,8 @@ totpErrorContact=If the OTP code was entered correctly and submitted before it r
 updatePasswordTitle=Set new password
 webauthn-error-register-verification=Your hardware token could not be registered. Contact <a href="mailto:tdr@nationalarchives.gov.uk">tdr@nationalarchives.gov.uk</a>.
 webauthn-error-api-get=Sign in with your hardware token failed. Contact <a href="mailto:tdr@nationalarchives.gov.uk">tdr@nationalarchives.gov.uk</a>.
+# AYR Keycloak template message overrides
+ayrBetaBannerInfo=Help us to improve this service by completing our
+ayrBetaBannerLink=feedback survey (opens in new tab).
+ayrHeader=Transfer and Access Your Records
+ayrPasswordReset=Reset your password
```
[the remark this answers is not in the document]
Comment (colinbowen) [reply themes/tdr/login/messages/messages_en.properties]: Added line break

File: themes/tdr/login/template.ftl [hunk 4]
```diff
@@ -78,6 +87,28 @@
...
+            </div>
+            <div class="govuk-header__content govuk-header__content--ayr">
+                Delivered by
+                <a href="https://www.nationalarchives.gov.uk/"
+                   class="govuk-header__link govuk-header__link--ayr">The
+                National Archives</a>
+            </div>
+        </div>
+    </header>
+
```
[the remark this answers is not in the document]
Comment (colinbowen) [reply themes/tdr/login/template.ftl]: Removed this line break

File: themes/tdr/login/messages/messages_en.properties [hunk 5]
```diff
@@ -43,3 +43,8 @@ totpErrorContact=If the OTP code was entered correctly and submitted before it r
 updatePasswordTitle=Set new password
 webauthn-error-register-verification=Your hardware token could not be registered. Contact <a href="mailto:tdr@nationalarchives.gov.uk">tdr@nationalarchives.gov.uk</a>.
 webauthn-error-api-get=Sign in with your hardware token failed. Contact <a href="mailto:tdr@nationalarchives.gov.uk">tdr@nationalarchives.gov.uk</a>.
+# AYR Keycloak template message overrides
```
[the remark this answers is not in the document]
Comment (colinbowen) [reply themes/tdr/login/messages/messages_en.properties]: Renamed as requested.

File: themes/tdr/css-src/sass/main.scss [hunk 6]
```diff
@@ -16,6 +16,9 @@ $govuk-fonts-path: '../fonts/';
 @import 'pagination';
 @import 'card';
 
+// AYR STYLES
```
[the remark this answers is not in the document]
Comment (colinbowen) [reply themes/tdr/css-src/sass/main.scss]: Moved the header styles into _header.scss

File: themes/tdr/login/login-reset-password.ftl [hunk 7]
```diff
@@ -1,9 +1,14 @@
...
+<#assign ayrPasswordResetPageTitle = msg("ayrPasswordReset")>
```
[the remark this answers is not in the document]
Comment (colinbowen) [reply themes/tdr/login/login-reset-password.ftl]: Renamed as requested

File: themes/tdr/login/template.ftl [hunk 8]
```diff
@@ -78,6 +87,28 @@
         </div>
     </header>
     <#-- End TDR Template header -->
+    <#else>
+    <#-- Start AYR Template header -->
```
[the remark this answers is not in the document]
Comment (colinbowen) [reply themes/tdr/login/template.ftl]: As above

File: themes/tdr/css-src/sass/_ayr.scss [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+        margin-bottom: 0.8rem;
+    }
+
+    @include on-mobile {
+        margin-right: 0;
+        font-size: 0.8rem;
+        display: inline-block;
+    }
+}
+
```
[the remark this answers is not in the document]
Comment (colinbowen) [reply themes/tdr/css-src/sass/_ayr.scss]: Removed as requested

File: themes/tdr/css-src/sass/_ayr.scss [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+        @include on-mobile {
+            padding-bottom: 1rem;
+            flex-direction: column;
+            align-items: center;
+            text-align: center;
+            padding-top: 0;
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (colinbowen) [reply themes/tdr/css-src/sass/_ayr.scss]: File has been deleted and moved to _header.scss

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks.

@anthonyhashemi / @colinbowen Before you merge these changes can you or someone else on AYR inform relevant stakeholders that the shared login page are going to be released onto TDR intg:
- Emma Peacock;
- Kirsten Arnold;
- AYR business owners

Manual testing of the login pages should then be undertaken to make sure they work correctly on intg. Staging should also be checked when this code is deployed to ensure the existing TDR login pages are still displayed and working correctly.

**Pull request merged**